sql-cli/tests/test___main__.py

```diff
@@ -245,6 +245,25 @@ def test_generate_invalid(workflow_name, message, initialised_project_with_tests
     assert message in result.stdout
 
 
+def test_generate_with_output_dir(initialised_project):
+    result = runner.invoke(
+        app,
+        [
+            "generate",
+            "example_basic_transform",
+            "--project-dir",
+            initialised_project.directory.as_posix(),
+            "--output-dir",
+            initialised_project.directory.as_posix(),
+        ],
+    )
+    expected_dag_file_path = initialised_project.directory / "example_basic_transform.py"
+    assert expected_dag_file_path.exists()
+    message = "has been successfully generated."
+    assert message in result.stdout
+    assert not result.exit_code
```

Comment:
```suggestion
    assert result.exit_code == 0
```
I think this shows more clearly that you expect it to succeed.